src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -446,58 +446,63 @@ bool StatefulReader::processDataMsg(
             logInfo(RTPS_MSG_IN,
                     IDSTRING "Trying to add change " << change->sequenceNumber << " TO reader: " << getGuid().entityId);
 
-            // Ask the pool for a cache change
-            CacheChange_t* change_to_add = nullptr;
-            if (!change_pool_->reserve_cache(change_to_add))
+            size_t unknown_missing_changes_up_to = pWP ? pWP->unknown_missing_changes_up_to(change->sequenceNumber) : 0;
+            if (mp_history->can_change_be_added_nts(change, unknown_missing_changes_up_to))
```

Comment:
Before this PR, when `change_received` fails, `false` is returned. But now, if `can_change_be_added_nts` fails, `true` is returned. Is this correct?

Reply:
Fixed on e17467fc61f10f5f4b1c1e38852d72123cf1bd3a